packages/app/src/hooks/useChartConfig.tsx

```diff
@@ -16,6 +17,25 @@ import { IS_MTVIEWS_ENABLED } from '@/config';
 import { buildMTViewSelectQuery } from '@/hdxMTViews';
 import { getMetadata } from '@/metadata';
 
+export const replaceBracketQuery = (query: any) => {
+  // sql-formatter throw error when query include brackets, ex: ResourceAttributes['telemetry.sdk.language']
+  // feel free to remove this parser once formatter support clickhouse dialect.
+  if (!_.isString(query)) return query;
+  // Replace key[value] to arrayElement(key, value)
+  // edge case:
+  // `key`[value], from lucene key.value
+  // key[number value], from array
+  // key[value1.value2._value3]
+  return query.replace(
+    /`?(\w+)`?\[(?:'([^']+)'|(\d+))\]/g,
+    (match, key, strValue, numValue) => {
+      return numValue !== undefined
+        ? `arrayElement(${key}, ${numValue})` // No quotes for numbers
+        : `arrayElement(${key}, '${strValue}')`; // Keep quotes for strings
```

Comment:
I think this hurts the sql readability. Ideally we can make the formatter work with bracket